src/DotNetWorker.Core/Logging/BindingLoggerHostedService.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace  Microsoft.Azure.Functions.Worker.Logging
+{
+    /// <summary>
+    /// Service responsible for logging the binding type that has been resolved for each function.
+    /// This log only occurs once 10 seconds the worker has been initialized to avoid cold start
+    /// impact.
+    /// </summary>
+    public class BindingLoggerHostedService : IHostedService, IDisposable
+    {
+        private readonly ILogger _logger;
+        private readonly Timer _timer;
+        private readonly int _due = 10000; // 10 seconds
+        private bool _disposed;
+
+        /// <summary>
+        /// TODO: should we be using IWorkerDiagnostics?
+        /// </summary>
+        public BindingLoggerHostedService(ILogger<BindingLoggerHostedService> logger)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
```

Comment:
@fabiocav wondering if I should be using `WorkerDiagnostics` here instead of logger or if there is something else I should be using